chainconfig/chainconfig_routine.go

```diff
@@ -85,6 +85,23 @@ func (cc *ChainConfigRoutine) run() {
 			} else {
 				// NOTE: errors are logged by the client, no need to log again
 				_ = chainConfigClient.VoteToEnablePendingFeatures(cc.buildNumber)
+				if !isBuildNumberEqual {
+					validatorInfo, err := chainConfigClient.GetValidatorInfo()
+					if err != nil {
+						cc.logger.Error("Failed to retreive build number", "err", err)
+					} else {
+						buildNumber := validatorInfo.Validator.GetBuildNumber()
```

Comment:
There is a chance that `validatorInfo.Validator` is nil. We need to check this before calling  `validatorInfo.Validator.GetBuildNumber()`